File: Bailaconmigo/src/main/java/org/example/bailaconmigo/DTOs/RegistrationReportDto.java
package org.example.bailaconmigo.DTOs;

import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.Map;

@Data
public class RegistrationReportDto {
    private LocalDate startDate;
    private LocalDate endDate;
    private LocalDateTime generatedAt;

    private int totalRegistrations;
    private int confirmedRegistrations;
    private int pendingRegistrations;
    private int cancelledRegistrations;

    private BigDecimal periodRevenue;
    private Map<LocalDate, Long> registrationsByDay;
}

File: frontend/src/app/models/registration-report-dto.ts
export interface RegistrationReportDto {
  startDate: string;
  endDate: string;
  generatedAt: string;
  totalRegistrations: number;
  confirmedRegistrations: number;
  pendingRegistrations: number;
  cancelledRegistrations: number;
  periodRevenue: number;
  registrationsByDay: Record<string, number>;
}
